operators/dream_texture.py

```diff
@@ -80,13 +92,7 @@ def execute(self, context):
             )
 
         # Setup the progress indicator
-        def step_progress_update(self, context):
-            if hasattr(context.area, "regions"):
-                for region in context.area.regions:
-                    if region.type == "UI":
-                        region.tag_redraw()
-            return None
-        bpy.types.Scene.dream_textures_progress = bpy.props.IntProperty(name="", default=0, min=0, max=generated_args['steps'], update=step_progress_update)
```

Comment:
I've found it's more performant to put similar functionality within the step callback. Strangely changing this to an empty function would still allow the UI to redraw. I assume it was redrawing more than necessary for seeing the progress bar. Allows for generating about 15%-20% faster in my experience.